viewing_party/party.py

```diff
@@ -1,23 +1,471 @@
+    # check the title for each element
+        if title == movie["title"]:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            print(user_data)  
+            return user_data
+    if title not in user_data["watchlist"]:
+        return user_data
+            
+# ========================================= wave 01- # 4. watch movie ========== SJ
+
+    # user_data= {
+    #     "watchlist": [{"title": "A movie title", "genre": "A movie genre", "rating": "soem number"}],
+        
+    #     "watched" : [{"title": "A movie title", "genre": "A movie genre", "rating": "soem number"}] 
+    #     }
+    
+    # user_data is a dict that contains 2 keys: watchlist and watched
+    
+    # movie is a ditionary format contained in both watched and watchlist
+    
+    # each movie dictionary has a key named "title"
+    
+    # the parameter title is a value, associated with the key "title"
+# def watch_movie(user_data, title):
+    
+#     for movie in user_data["watchlist"]: 
+        
+#         if title == movie["title"]:
+#             # remove that movie dictionary from the watchlist
+#             user_data["watchlist"].remove(movie)
+
+#             # add the movie dictionary to the the watched
+#             user_data["watched"].append(movie)
+            
+#             #then return user data
+#             # print(user_data)
+#             return user_data
+            
+#             # else, if the title is not a movie in the user's watchlist, 
+#             # return the user_data
+#     else: 
+#         return user_data
+            
+        
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+# ========================================= wave 02- # 1. get_watched_avg_rating ========== RC
 
+# def get_watched_avg_rating(user_data):
+#     # This represents that the user has a list of watched movies
+#     watched = user_data.get("watched", [])
+#     # The average rating of an empty watched list is `0.0`
+#     if not watched:
+#         return 0.0
+#     # Calculate the average rating of all movies in the watched list
+#     total_ratings = sum(movie.get("rating", 0.0) for movie in watched)
+#     # return the average rating
+#     avg_rating = total_ratings / len(watched)
+#     return avg_rating
+
+# ========================================= wave 02- # 2. get_most_watched_genre ========== RC
+
+def get_most_watched_genre(user_data):
+    # keep the var the same
+    watched = user_data["watched"]
+    # if movie title not watched, return none
+    if not watched:
+        return None
+    
+    # iterate thru each "genre" key's value for the entire length of nested dictionary
+    genres_watched = [watched[i]["genre"] for i in range(len(watched))]
+    
+    # get the most watched genre with max function
+    return max(genres_watched, key=genres_watched.count)

```

Comment:
This function takes a nice approach! Great job handling the logic with the use of `max` and list comprehension 👍🏾